ckan/lib/jinja_extensions.py

```diff
@@ -261,20 +273,61 @@ def make_call_node(*kw: Any) -> Any:
         return nodes.Output([make_call_node()]).set_lineno(tag.lineno)
 
 
-class SnippetExtension(BaseExtension):
+class SnippetExtension(ext.Extension):
     ''' Custom snippet tag
 
     {% snippet <template_name> [, <fallback_template_name>]...
                [, <keyword>=<value>]... %}
 
-    see lib.helpers.snippet() for more details.
+    renders <template_name> with <keyword>=<value> as local variables
     '''
 
-    tags = set(['snippet'])
+    tags = {'snippet'}
+
+    def parse(self, parser: Parser):
```

Comment:
I must confess that I don't fully grasp this code. Would this be a fair summary?
Here we parse the `{% snippet 'path/to/snippet.html', arg1=test %}` string, and turn it into a node tree equivalent to:

```
{% with %}
    {% import ... as snippet %}
    {{ snippet(...) }}
{% endwith %}
```

And then this will be picked up in the `CkanFileSystemLoader.parse()` method above which will turn it into :

```
{% macro snippet(...) %}
    {% include ... %}
{% endmacro %}```

Which is what will be finally rendered by Jinja. Or is this the other way around?
Sorry I find this a bit confusing

Reply:
That's right. The snippet `with`/`import` node tree references a specially named wrapper template `original_name§arg1,arg2.html` that gets generated by `CkanFileSystemLoader.get_source()` containing `macro`/`include` tags.

The imported wrapper template is necessary to create a separate namespace and convert `arg1`, `arg2`… into local variables for the snippet template.